### PR TITLE
Cleanup ProjectN references

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/pull/68151#issuecomment-1101686845

File: src/coreclr/nativeaot/System.Private.DisabledReflection/src/Internal/Reflection/RuntimeTypeInfo.cs [hunk 1]
```diff
@@ -187,7 +187,10 @@ protected override bool IsByRefImpl()
             return RuntimeAugments.IsByRefType(_typeHandle);
         }
 
-        protected override bool IsCOMObjectImpl() => throw new NotSupportedException(SR.Reflection_Disabled);
+        protected override bool IsCOMObjectImpl()
+        {
+            return false;
```
Comment (danmoseley) [inline src/coreclr/nativeaot/System.Private.DisabledReflection/src/Internal/Reflection/RuntimeTypeInfo.cs:192]: Unrelated fix ?

File: src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ArgIterator.cs [hunk 2]
```diff
@@ -1384,15 +1357,9 @@ private void ForceSigWalk()
...
-#if PROJECTN
-                    // On ProjectN ret buff arg is passed on the call stack as the top stack arg
-                    nSizeOfArgStack += _transitionBlock.PointerSize;
-#else
                     numRegistersUsed++;
-#endif
                 }
 
-#if PROJECTN
                 // DESKTOP BEHAVIOR - This block is disabled for x86 as the param arg is the last argument on .NET Framework x86.
```
Comment (danmoseley) [inline src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ArgIterator.cs:1396]: Intentionally took the #if block here?

**Review by danmoseley — APPROVED** [review]

File: src/coreclr/nativeaot/System.Private.DisabledReflection/src/Internal/Reflection/RuntimeTypeInfo.cs (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.DisabledReflection/src/Internal/Reflection/RuntimeTypeInfo.cs:192]: Yes, it is for symmetry with the other `IsComObjectImpl` implementation that I have removed `PROJECTN` ifdef from.

File: src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ArgIterator.cs (hunk 2 above)
Comment (jkotas) [reply src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ArgIterator.cs:1396]: Good catch! This was not intentional.

**Review by MichalStrehovsky — APPROVED** [review]

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs [hunk 3]
```diff
@@ -109,9 +109,9 @@ private static void ValidateElementType(Type elementType)
 
         public void Initialize()
         {
-            // Project N port note: On the desktop, this api is a nop unless the array element type is a value type with
-            // an explicit nullary constructor. Such a type cannot be expressed in C# so Project N does not support this.
-            // The ILC toolchain fails the build if it encounters such a type.
+            // This api is a nop unless the array element type is a value type with an explicit nullary constructor.
+            // Such a type could not be expressed in C# up until recently.
+            // The AOT toolchain fails the build if it encounters such a type.
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs:114]:
```suggestion
            // TODO: Implement this
```

**Comment by jkotas** [comment]
The failures are:
#67761
#67349
#67821
